### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Leen-odeh3**

### CODE REVIEW

File: RestaurantReservation.Db/Abstracts/ICustomerRepository.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿
+
+using RestaurantReservation.Db.Entities;
+
+namespace RestaurantReservation.Db.Abstracts
+{
+    public interface ICustomerRepository : IRepository<Customer>
+    {
+    }
+}
```
Comment (ahmad-m-abbas) [inline RestaurantReservation.Db/Abstracts/ICustomerRepository.cs:10]:
```suggestion
namespace RestaurantReservation.Db.Abstracts;
public interface ICustomerRepository : IRepository<Customer>
{
}
```

File: RestaurantReservation.Db/Configuration/EmployeeConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+namespace RestaurantReservation.Db.Configuration
+{
+    public class EmployeeConfiguration : IEntityTypeConfiguration<Employee>
+    {
+        public void Configure(EntityTypeBuilder<Employee> builder)
+        {
+            builder.Property(e => e.FirstName).IsRequired().HasMaxLength(50);
+            builder.Property(e => e.LastName).IsRequired().HasMaxLength(50);
+            builder.Property(e => e.Position).IsRequired();
+            builder.HasOne(e => e.Restaurant)
```
Comment (ahmad-m-abbas) [inline RestaurantReservation.Db/Configuration/EmployeeConfiguration.cs:14]: Format to vertical calls

File: RestaurantReservation.Db/Entities/Customer.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+namespace RestaurantReservation.Db.Entities
+{
+    public class Customer
+    {
+        public int Id { get; set; }
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+        public string Email { get; set; }
+        public string PhoneNumber { get; set; }
+        public ICollection<Reservation>? Reservations { get; set; } = new List<Reservation>();
```
Comment (ahmad-m-abbas) [inline RestaurantReservation.Db/Entities/Customer.cs:13]:
```suggestion
        public virtual ICollection<Reservation>? Reservations { get; set; }
```

File: RestaurantReservation.Db/Entities/Reservation.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+using System.Collections.Generic;
+
+namespace RestaurantReservation.Db.Entities
+{
+    public class Reservation
+    {
+        public int Id { get; set; }
+        public DateTime ReservationDate { get; set; }
+        public int PartySize { get; set; }
+        public int? CustomerId { get; set; }
```
Comment (ahmad-m-abbas) [inline RestaurantReservation.Db/Entities/Reservation.cs:12]: can a foreign key be null?

**Review by ahmad-m-abbas — CHANGES_REQUESTED** [review]

File: RestaurantReservation.Db/Abstracts/ICustomerRepository.cs [hunk 5]
```diff
@@ -2,6 +2,6 @@
 
 namespace RestaurantReservation.Db.Abstracts;
 
-public interface ICustomerRepository : IRepository<Customer>
-{
-}
+    public interface ICustomerRepository : IRepository<Customer>
+    {
+    }
```
Comment (ahmad-m-abbas) [inline RestaurantReservation.Db/Abstracts/ICustomerRepository.cs:7]: ?

Comment (Leen-odeh3) [reply RestaurantReservation.Db/Abstracts/ICustomerRepository.cs]: delete it ? Not used anywhere

**Review by ahmad-m-abbas — APPROVED** [review]

**Comment by ahmad-m-abbas** [comment]
@Leen-odeh3 resolve conflicts

**Pull request merged**